core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -439,4 +439,40 @@ public static MetricsReporter loadMetricsReporter(String impl) {
 
     return reporter;
   }
+
+  /**
+   * Load a custom ClientPool implementation.
+   *
+   * <p>The ClientPool must have a no-arg constructor. {@link ClientPool#initialize(Map, Object)} is
+   * called to complete the initialization.
+   *
+   * @param impl ClientPool implementation full class name
+   * @param properties catalog properties
+   * @param conf hadoop configuration if needed
+   * @return initialized ClientPool object
+   * @throws IllegalArgumentException if no-arg constructor not found or error during initialization
+   */
+  public static <C, E extends Exception> ClientPool<C, E> loadClientPool(
+      String impl, Map<String, String> properties, Object conf) {
+    Preconditions.checkNotNull(
+        impl, "Cannot initialize custom ClientPool, impl class name is null");
```

Comment:
nit, do you think we can also have a better word in the error message than 'impl', ie use the actual property constant name, and we can pass it to Preconditions via the .checkNonNull(errorMsgTemplate, errorMsgArgs)?